addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -376,6 +373,17 @@ export class ProductScreen extends Component {
         });
     }
 
+    getProductsByCategory(categoryId) {
+        let products = this.pos.models['product.product'].getBy('pos_categ_ids', categoryId) || [];
+        let childCategories = this.pos.models['pos.category'].get(categoryId).child_id;
+
+        for (const { id } of childCategories) {
+            products = products.concat(this.getProductsByCategory(id));
+        }
+
+        return products;
```

Comment:
There might be a need to deduplicate the products returned here. It's possible that a product belong to two categories in which one of the category is ancestor of the other.

Reply:
@caburj I updated the code, and used Set to prevent duplication